### ISSUE TITLE
No system menu in the navigation bar

### ISSUE THREAD

**Issue by malnvenshorn** [foosel/OctoPrint#1495]
Hi,

after adding the commands for shutdown, reboot and restarting OctoPrint I wondered where I can activate those commands. After some search I found out that there should be a pulldown menu in the navigation bar, but this menu is missing. All I can see are settings, the user menu and the announcements icon. Of course the specific user has admin rights. I'm using version 1.2.15 of OctoPrint and opened the web frontend with Firefox 48.0.2 and InternetExplorer 11. Same result with both browsers.

**Comment by GitIssueBot** [comment]
Hi @malnvenshorn,

It looks like there is some **information missing** from your bug report that will be needed in order to solve the problem. Read the [Contribution Guidelines](https://github.com/foosel/OctoPrint/blob/master/CONTRIBUTING.md) which will provide you with a template to fill out here so that your bug report is ready to be investigated (I promise I'll go away then too!).

If you did not intend to report a bug but wanted to **request a feature or brain storm** about some kind of development, please take special note of the title format to use as described in the [Contribution Guidelines](https://github.com/foosel/OctoPrint/blob/master/CONTRIBUTING.md).

**Please do not abuse the bug tracker as a support forum** - if you have a question or otherwise need some kind of help or support refer to the [Mailinglist](https://groups.google.com/group/octoprint) or the [G+ Community](https://plus.google.com/communities/102771308349328485741) instead of here.

Also **make sure you are at the right place** - this is the bug tracker of the official version of OctoPrint, not the Raspberry Pi image OctoPi nor any unbundled third party OctoPrint plugins or unofficial versions. Make sure too that you have **read through the [Frequently Asked Questions](https://github.com/foosel/OctoPrint/wiki/FAQ)** and searched the [**existing tickets**](https://github.com/foosel/OctoPrint/search?q=&ref=cmdform&type=Issues) for your problem - try multiple search terms please.

I'm marking this one now as needing some more information. Please understand that if you do not provide that information within the next two weeks (until 2016-09-27 20:00 UTC) I'll close this ticket so it doesn't clutter the bug tracker. This is nothing personal, so please just be considerate and help the maintainers solve this problem quickly by following the guidelines linked above. Remember, the less time the devs have to spend running after information on tickets, the more time they have to actually solve problems and add awesome new features. Thank you!

Best regards,
~ Your friendly GitIssueBot

PS: I'm just an automated script, not a human being, so don't expect any replies from me :) Your ticket is read by humans too, I'm just not one of them.

**Comment by foosel** [comment]
If you haven't, try restarting OctoPrint after adusting config.yaml. If you have already done that and it didn't help, please get in touch on either the mailing list or the g+ community. When you do, also include your config.yaml file or at least the snippet you copy-pasted into it. It's sadly impossible to help you without having access to that.

**Comment by malnvenshorn** [comment]
Yes I already restarted OctoPrint and I haven't touched the config.yaml file. I set up everything inside the web interface under Settings > Server > Commands. The generated config.yaml file can be found here: https://paste.debian.net/823918/

**Comment by foosel** [comment]
Ah, that explains it. While current `devel` already uses the commands from Settings > Server > Commands to also to populate the system menu, current stable releases still need those things [configured separately](http://docs.octoprint.org/en/master/configuration/config_yaml.html#system) and the commands are only used for internal functions such as the software update process there.

Try:

``` yaml
system:
  actions:
  - name: Restart OctoPrint
    action: restart
    command: sudo systemctl restart octoprint.service
    confirm: You are about to restart OctoPrint.
  - name: Reboot the server
    action: reboot
    command: sudo reboot
    confirm: You are about to reboot the whole server.
  - name: Shutdown the server
    action: shutdown
    command: sudo poweroff
    confirm: You are about to shutdown the whole server.
```

in your `config.yaml` and things should work on 1.2.x as well. You can also leave out the `confirm` lines if you don't want a confirmation dialog.

**Comment by ctgreybeard** [comment]
I'm running 1.2.16rc2 and I tried adding those commands under Settings->Server->Commands but they do not appear in the System menu, even after restarting the service.

**Comment by foosel** [comment]
`devel` is 1.3.0dev. 1.2.16rc2 fits the above mentioned version pattern 1.2.x and still needs the `system.actions` stanzas in config.yaml as outlined above.

**Comment by ctgreybeard** [comment]
Ah! Thanks.

**Comment by jcdammeyer** [comment]
I've attached my Replicape config.yaml file.   I now do have the System menu up there but none of the actions have any effect.  It doesn't shut down or reboot.  Also I've noticed that there's no direct link with the fields in the server menu to the System menu selections so there are really two issues.  

if I perform a sudo poweroff on the command line a short while later my Beagle is off.  But from the menu nothing.  Same with the -r restart or restarting octoprint.

I've looked at the web source for yaml syntax and there's no linkage so is that done in the Python application then  that links the commands in the fields to the menu entries?
Thanks,
[Uploading Replicape-config.yaml.txt…]()
